app-modules/service-management/src/Observers/ChangeRequestTypeObserver.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/*
+<COPYRIGHT>
+
+    Copyright © 2016-2024, Canyon GBS LLC. All rights reserved.
+
+    Advising App™ is licensed under the Elastic License 2.0. For more details,
+    see https://github.com/canyongbs/advisingapp/blob/main/LICENSE.
+
+    Notice:
+
+    - You may not provide the software to third parties as a hosted or managed
+      service, where the service provides users with access to any substantial set of
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Advising App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+namespace AdvisingApp\ServiceManagement\Observers;
+
+use AdvisingApp\ServiceManagement\Models\ChangeRequestType;
+
+class ChangeRequestTypeObserver
+{
+    public function deleted(ChangeRequestType $changeRequestType): void
+    {
+        if ($changeRequestType->changeRequests()->doesntExist()) {
+            $changeRequestType->forceDelete();
+        }
+    }
```

Comment:
I am not sure about doing this here and similarly with the other observers.

I think it would be preferred that delete and forceDelete remain as separate actions in the system.

If we want to improve the DeleteAction to present the ability to trigger forceDelete that is fine. But delete should be possible without force delete regardless of the relationship.

Reply:
I think I would normally agree, but the wording of the request from Product was:
```
If unused, introduce ability to delete.

If used, add ability to archive.
```
As we discussed earlier, I read this as if used soft delete, otherwise force delete. If we don't have this distinction then there's no meaningful difference between deleting a used or unused record. 

We could add a checkbox to the delete action form that switch to a soft delete, but that could still present the same flow problem, i.e., the user deletes a record, forgets about it, then tries to recreate it and runs into the unique constraint. We could have the query exclude deleted records, but then what's the point in having a bunch of hidden unused records? Also, unless we are going to introduce the custom checkbox action everywhere as our deletion standard then I don't think we should do it here.

The double action flow to force delete something later also isn't great because it depends on the user first remembering that there is something to delete, changing the filter to view it, going to the edit page, finally clicking force delete. The seems far less likely after the record is hidden.

Another option would be to pull the force delete action forward and swap the presented action based on whether the record is used or unused, but that feels like a harder pivot from the Filament paradigm than the other suggestions. Truth be told I don't know which way Product would lean. I can ask for more clarification unless you have a strong opinion?

Reply:
Right now, my gut is saying this doesn't belong in the observer. Nowhere else in the application do we force delete during a delete action, and I feel that this creates an unexpected and hidden backend consequence. While at the same time removing the ability to archive an unused record if you wanted to.

I would lean more towards modifying the front end to have a more clear distinction in action.

To modify the delete action, if the record is unused, add something like a check box to force delete. Otherwise, soft delete.
Then, you can also add/modify an action to force delete an unused trashed record later.

That gives the User much more control and information in the situation while not creating singular and unexpected results in the backend.

For the concern of then there being a hidden record. I would say we catch the UniqueConstraintException that would be thrown or check for this scenario and notify the User that there is a duplicate archived record.

@dgoetzit and @danharrin, what do y'all think?